internal/plugin/provider.go

```diff
@@ -134,6 +135,7 @@ func (p *AivenProvider) DataSources(context.Context) []func() datasource.DataSou
 	// List of data sources that are currently available in the provider.
 	dataSources := []func() datasource.DataSource{
 		organization.NewOrganizationDataSource,
+		externalidentity.NewExternalIdentityDataSource,
```

Comment:
Removing L138 and mark this as a beta resource:

```golang
func (p *AivenProvider) DataSources(context.Context) []func() datasource.DataSource {
	// List of data sources that are currently available in the provider.
	dataSources := []func() datasource.DataSource{
		organization.NewOrganizationDataSource,
	}

	// Add to a list of data sources that are currently in beta.
	if util.IsBeta() {
		betaDataSources := []func() datasource.DataSource{
			externalidentity.NewExternalIdentityDataSource,
		}
		dataSources = append(dataSources, betaDataSources...)
	}

	return dataSources
}
```

You'll need to run `make generate` after these changes